C error when assert with optional

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:**  V 0.2.2 39c376b
**OS:** Windows 10, Ubuntu 20

**What did you do?**
 ```v
import os
fn main(){
	assert os.truncate("testfile.txt", 6666) or { panic(err) }
}
```
Compiled above with `v -cc [msvc, gcc, tcc, clang] .`
**What did you expect to see?**
Compiles or V error
 
**What did you see instead?**
```
==================
<..>.10499860463605147662.tmp.c:13562: warning: assignment from incompatible pointer type
<..>.10499860463605147662.tmp.c:13729: error: ';' expected (got "}")
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

The bad C-Code:
```c
VV_LOCAL_SYMBOL void main__main(void) {
	Option_void _t1118 = os__truncate(_SLIT("testfile.txt"), 6666U);
	if (_t1118.state != 0 && _t1118.err._typ != _IError_None___index) {
		IError err = _t1118.err;
		(v_panic(IError_str(err)))}
 	// assert
	if (!()) {
		VAssertMetaInfo v_assert_meta_info__t1119 = {0};
		v_assert_meta_info__t1119.fpath = _SLIT("vshred.v");
		v_assert_meta_info__t1119.line_nr = 19;
		v_assert_meta_info__t1119.fn_name = _SLIT("main.main");
		v_assert_meta_info__t1119.src = _SLIT("main.os.truncate('testfile.txt', 6666)");
		v_assert_meta_info__t1119.op = _SLIT("call");
		__print_assert_failure(&v_assert_meta_info__t1119);
		v_panic(_SLIT("Assertion failed..."));
	}
}
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -4064,7 +4064,7 @@ fn (mut c Checker) stmt(node ast.Stmt) {
 fn (mut c Checker) assert_stmt(node ast.AssertStmt) {
 	cur_exp_typ := c.expected_type
 	assert_type := c.check_expr_opt_call(node.expr, c.expr(node.expr))
-	if assert_type != ast.bool_type_idx && assert_type != ast.void_type_idx {
+	if assert_type != ast.bool_type_idx {
 		atype_name := c.table.get_type_symbol(assert_type).name
 		c.error('assert can be used only with `bool` expressions, but found `$atype_name` instead',
 			node.pos)

--- a/vlib/v/checker/tests/assert_optional_err.out
+++ b/vlib/v/checker/tests/assert_optional_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/assert_optional_err.vv:4:9: error: assert can be used only with `bool` expressions, but found `void` instead
+    2 |
+    3 | fn main(){
+    4 |     assert os.truncate("testfile.txt", 6666) or { panic(err) }
+      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    5 | }

--- a/vlib/v/checker/tests/assert_optional_err.vv
+++ b/vlib/v/checker/tests/assert_optional_err.vv
@@ -1,0 +1,5 @@
+import os
+
+fn main(){
+	assert os.truncate("testfile.txt", 6666) or { panic(err) }
+}

--- a/vlib/v/parser/parser.v
+++ b/vlib/v/parser/parser.v
@@ -685,7 +685,7 @@ pub fn (mut p Parser) stmt(is_top_level bool) ast.Stmt {
 			pos.update_last_line(p.prev_tok.line_nr)
 			return ast.AssertStmt{
 				expr: expr
-				pos: pos
+				pos: pos.extend(p.tok.position())
 				is_used: p.inside_test_file || !p.pref.is_prod
 			}
 		}

